qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java

```diff
@@ -0,0 +1,288 @@
+    private static final String MAXIMUM_OCCURS = RandomStringUtils.randomNumeric(5);
+    private static final String DATATYPE_CONFIGURATION = RandomStringUtils.randomAlphabetic(8);
+    private static final String LOCATION_HANDLER_CONFIGURATION = RandomStringUtils.randomAlphabetic(8);
+    private static final String LOCATION_NAME = RandomStringUtils.randomAlphabetic(5);
+    private static final String LOCATION_TAG = RandomStringUtils.randomAlphabetic(5);
+    private static final String LOCATION_DESCRIPTION = RandomStringUtils.randomAlphabetic(8);
+
+    @Before(RunTest.HOOK.SELENIUM_LOCATION_MANAGEMENT)
+    public void visitDashboard() {
+        initiateWithLogin();
+    }
+
+    @After(RunTest.HOOK.SELENIUM_LOCATION_MANAGEMENT)
+    public void destroy() {
+        quit();
+    }
+
+    @Given("a user clicks on the configure metadata link from the home page")
+    public void userClicksOnConfigureMetadataLink() {
+        configureMetadataPage = homePage.goToConfigureMetadata();
+    }
+
+    @Then("the system loads configure metadata link dashboard")
+    public void systemLoadsMetadataDashboard() {
+        assertTrue(textExists("Configure Metadata"));
+    }
+
+    @When("a user clicks on the Manage Location Attribute Types")
+    public void userClicksOnManageLocationAttributeTypes() {
+        manageLocationAttributeTypesPage = configureMetadataPage.goToManageLocationAttributeTypesPage();
+    }
+
+    @Then("the system loads the manage location attribute type page")
+    public void systemLoadsTheManageLocationAttributeTypePage() {
+        assertTrue(textExists("Manage Location Attribute Types"));
+    }
+
+    @And("a user clicks on the add new location attribute type")
+    public void userClicksOnAddNewLocationAttributeType() {
+        manageLocationAttributeTypesPage.goToAddNewLocationAttributeTypeForm();
+    }
+
+    @And("a user fills the add new location attribute type form")
+    public void userFillsAddNewLocationAttributeTypeForm() {
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeName(LOCATION_ATTRIBUTE_TYPE_NAME);
+        manageLocationAttributeTypesPage.enterLocationAttributeTypeDescription(LOCATION_ATTRIBUTE_TYPE_DESCRIPTION);
+        manageLocationAttributeTypesPage.enterMinimumOccurs(MINIMUM_OCCURS);
+        manageLocationAttributeTypesPage.enterMaximumOccurs(MAXIMUM_OCCURS);
+        manageLocationAttributeTypesPage.selectDatatype("Location Datatype");
+        manageLocationAttributeTypesPage.enterDatatypeConfiguration(DATATYPE_CONFIGURATION);
+        manageLocationAttributeTypesPage.selectPreferredHandler("Location Field Gen Datatype Handler");
+        manageLocationAttributeTypesPage.enterHandlerConfiguration(LOCATION_HANDLER_CONFIGURATION);
+    }
+
+    @Then("the user saves the form")
+    public void userSavesForm() {
+        manageLocationAttributeTypesPage.clickOnSaveButton();
+        assertTrue(textExists("Manage Location Attribute Types"));
```

Comment:
Here, checking for the text "Manage Location Attribute Types" won't be enough to ensure that the form is saved correctly. Because it is always there on the page.

<img width="483" alt="image" src="https://user-images.githubusercontent.com/27498587/228474252-a6bce76d-65e3-4996-89fb-8e7ccb601fcf.png">

It would be better to check whether the name of the created `Location Attribute Type` is on the table.